src/lib/Libifl/pbsD_submit.c

```diff
@@ -219,6 +218,15 @@ __pbs_submit(int c, struct attropl  *attrib, char *script, char *destination, ch
 			extend = EXTEND_OPT_IMPLICIT_COMMIT;
 	}	
 
+	c = random_svr_conn;	
+	if (msvr && destination) {
+		/* Reached here means job is submitted to non default queue */
+		int start;
+
+		if ((start = get_shard_obj_location_hint(destination, MGR_OBJ_RESV)) != -1)
```

Comment:
Please add a comment here explaining that since this could be a reservation queue and reservation queues are not shared, we try to find out which server this queue resides on.